### PR TITLE
101 housing

### LINKED ISSUE #94: Mean Salary (ASHE)
**Summary**
- Date: 2011 - 2017
- Sources: APS

**Progress**
- [x] Scoped
- [x] Collected
- [x] Uploaded
- [x] Raw Schema
- [x] Processed
- [x] Processed Schema

**Challenges and Observations**

### LINKED ISSUE #101: House affordability
**Summary**
- Date: 
- Sources:

**Progress**
- [x] Scoped
- [x] Collected
- [x] Extracted
- [x] Uploaded
- [ ] Raw Schema
- [ ] Processed
- [ ] Processed Schema

**Challenges and Observations**

### PR DESCRIPTION
**Pull request by Juan-Mateos**
Closes #101  
Closes #94

### CODE REVIEW

File: data/processed/ashe_place/2020_01_30_ashe_lep_all_occupations.csv [hunk 1]
```diff
@@ -0,0 +1,191 @@
+,date_name,geography_name,geography_code,obs_value
```
Comment (mindrones) [inline data/processed/ashe_place/2020_01_30_ashe_lep_all_occupations.csv]: Please:
- remove the index
- rename the headers according to the schema template (*)
- remove the geography name column
- remove the date from the file name
- add a `data_date` to the schema

As these are LEP regions which we aren't supporting in the current setup:
- I'd use `year`, `lep_id, `lep_year_spec`, `value.id` (undocumented for now)
- I'd move this in a dir `data/processed/ashe_place/LEP/` for now

File: data/processed/housing/2020_01_29_nuts_house_prices.csv [hunk 2]
```diff
@@ -0,0 +1,370 @@
+,NUTS218CD,NUTS218NM,year,total_sales
```
Comment (mindrones) [inline data/processed/housing/2020_01_29_nuts_house_prices.csv]: Please:
- remove the index
- rename the headers according to the schema template
- remove the `geography_name` column
- add the `nuts_year_spec` column
- remove the date from the file name
- provide a schema
- add `data_date` to the schema

File: data/processed/ashe_place/2020_01_30_ashe_nuts_2_sci_tech.csv [hunk 3]
```diff
@@ -0,0 +1,273 @@
+,year,nuts_2,occupation,gross_annual_salary_median,nuts_2_codes
```
Comment (mindrones) [inline data/processed/ashe_place/2020_01_30_ashe_nuts_2_sci_tech.csv]: Please:
- remove the index
- rename the headers according to the schema template
- remove the geography name column
- remove the date from the file name
- provide a schema
- add `data_date` to the schema

Also this file seems to contain this indicators per each `occupation`, which is not the shape we discussed. Not sure how to approach this, as making a new fiel per each single `occupation` would not make sense I think for 2 reasons:
- these would seem different indicators
- as for UX, the user would see a lot of these as distinct links in the sidebar making it difficult to mentally group them

We would need to add a "category" column for this kind of situation, but this has to be discussed.

File: data/processed/house_afford/ratio_house_price_salary.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+api: the type of API to be used to fetch data: fetch
```
Comment (mindrones) [inline data/processed/house_afford/ratio_house_price_salary.yaml]: As this is an indicator derived from other indicators, for these fields I'd use objects like introduced in #134:

```yaml
api:
  - @ashe_nuts_2_sci_tech.api
  - @nuts_house_prices.api
api_documentation:
  - @ashe_nuts_2_sci_tech.api_documentation
  - @nuts_house_prices.api_documentation
endpoints:
  - @ashe_nuts_2_sci_tech.enpoint_static
  - @nuts_house_prices.enpoint_static
source:
  - @ashe_nuts_2_sci_tech.source
  - @nuts_house_prices.source
```

or alternatives

File: notebooks/dev/notes.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+## Lab notes
+
+* Need to distinguish between repos based on proprietary data which we access through DAPS and repos based on open data, where the whole pipeline can be shared in GitHub
```
Comment (mindrones) [inline notebooks/dev/notes.md]: Please remove this section and open a Github issue about it

File: notebooks/dev/notes.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+### ashe_place
+
+* We considered using information from the Nomis ASHE api to gross the wage bill in a location so we could recalculate at the NUTS2 level but this would have meant only being able to report average salaries in the NUTS2 areas, and additional complications obtaining the number of full time jobs data.
```
Comment (mindrones) [inline notebooks/dev/notes.md]: I'd consider remove the datasets that are not using NUTS2 and work on this later on as for this first iteration working on other kind of regions would be too complex given the contraints we have at the moment.

File: data/processed/house_afford/ratio_house_price_salary.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+description: We calculate house affordability in a NUTS2 region as the ratio between average house prices based on the House Pride index, and median salaries for science and technology occupations from ASHE. In order to calculate mean house prices at the NUTS2 level we needed to perform some transformations on the House Price Index data, which was only available at NUTS4. This is why we are labelling the indicator as experimental. Also note that the ASHE data was not available at the NUTS2 level from NOMIS so we had to use a data dump from ASHE. This has the advantage of being available at the NUTS2 level and by occupation, but cons around future maintenance.
+enpoint_static: http://publicdata.landregistry.gov.uk/market-trend-data/house-price-index-data/UK-HPI-full-file-2019-08.csv?utm_medium=GOV.UK&utm_source=datadownload&utm_campaign=full_fil&utm_term=9.30_16_10_19 (HPI) and https://www.ons.gov.uk/file?uri=/employmentandlabourmarket/peopleinwork/earningsandworkinghours/adhocs/009571annualsurveyofhoursandearningsasheestimatesofannualandhourlyearningsforindustryandoccupationbynuts2andnuts3uk2011to2017/k42forpublishing.zip (ASHE)
```
Comment (mindrones) [inline data/processed/house_afford/ratio_house_price_salary.yaml]: Move these to the correspondent schema.

Note that we need to strip the tracking token from the first link:
```
?utm_medium=GOV.UK&utm_source=datadownload&utm_campaign=full_fil&utm_term=9.30_16_10_19
```

(see #134)

**Review by mindrones — CHANGES_REQUESTED** [review]
first pass.

Please append [1] to the first PR comment (`#??` being the number of the issue relative to HPI, not sure which one it is), so that we add references PR <-> issues.

[1]
```
Closes #??
Closes #94
Closes #101
```

**Comment by Juan-Mateos** [comment]
Closes 101
Closes 94

File: data/processed/ashe_place/gbp_gross_median_salary_s_t.csv [hunk 8]
```diff
@@ -0,0 +1,273 @@
+year,nuts_id,nuts_year_spec,gbp_gross_median_salary_s_t
+2011,UKC1,2018,32961.0
```
Comment (mindrones) [inline data/processed/ashe_place/gbp_gross_median_salary_s_t.csv:2]: I think we need to run autonuts on these (2018 > 2011).

Also, not sure about the indicator name here, the fact that it ends in `_s_t` makes is pretty obscure to me.

File: data/processed/ashe_place/gbp_gross_median_salary_s_t.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+api: fetch
```
Comment (mindrones) [inline data/processed/ashe_place/gbp_gross_median_salary_s_t.yaml:1]: This needs to be update to the newer schema template

File: data/processed/house_afford/ratio_house_price_salary.yaml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  - @gbp_gross_median_salary_s_t.api
+  -https://www.gov.uk/government/publications/about-the-uk-house-price-index/about-the-uk-house-price-index#monthly-revision
+description: We calculate house affordability in a NUTS2 region as the ratio between average house prices based on the House Pride index, and median salaries for science and technology occupations from ASHE.
+enpoints: 
+  - @gbp_gross_median_salary_s_t.endpoint_static
+  - http://publicdata.landregistry.gov.uk/market-trend-data/house-price-index-data/UK-HPI-full-file-2019-08.csv?utm_medium=GOV.UK
+framework_group: place_potential
+is_experimental: True
+nuts_level: 2
+<<<<<<< Updated upstream
```
Comment (mindrones) [inline data/processed/house_afford/ratio_house_price_salary.yaml:12]: there are rebase conflict artifacts here

File: data/processed/house_afford/ratio_house_price_salary.yaml [hunk 11]
```diff
@@ -0,0 +1,34 @@
+api: fetch
```
Comment (mindrones) [inline data/processed/house_afford/ratio_house_price_salary.yaml:1]: This needs to be update to the newer schema template

**Review by mindrones — CHANGES_REQUESTED** [review]
some change required

**Comment by Juan-Mateos** [comment]
I have been reviewing this and found some issues with the salary data we are using to normalise housing prices to calculate affordability. Some of their NUTS categories seem to be non-standard ie they aggregated all London in two areas. I suggest we replace it with annual earnings by inhabitant from Eurostat (https://appsso.eurostat.ec.europa.eu/nui/show.do?dataset=nama_10r_2hhinc&lang=en) but not until the final iteration of the indicators.

**Comment by mindrones** [comment]
Closing temporarily as this issue is on hold

**Pull request closed without merging**